File: Year-2023/LeetCode-August-Challenge/Searcha2DMatrix.js
var searchMatrix = function(matrix, target) {
    let rows = matrix.length, cols = matrix[0].length;
    for(let i = rows - 1, j = 0; i >= 0 && j < cols;){
        if(target === matrix[i][j]) return true;
        else if(target > matrix[i][j]) j++;
        else i--;
    }
    return false;
};

let matrix = [[1,3,5,7],[10,11,16,20],[23,30,34,60]], target = 3;

console.log(searchMatrix(matrix, target));

File: Year-2023/LeetCode-August-Challenge/Searcha2DMatrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int row = matrix.length, cols = matrix[0].length;
        for(int i = row - 1, j = 0; i >= 0 && j < cols;){
            if(target == matrix[i][j]) return true;
            else if(target > matrix[i][j]) j++;
            else i--;
        }
        return false;
    }
}
public class Searcha2DMatrix{
    public static void main(String[] args) {
        int[][] matrix = {{1,4,7,11,15},{2,5,8,12,19},{3,6,9,16,22},{10,13,14,17,24},{18,21,23,26,30}};
        int target = 5;
        System.out.println(new Solution().searchMatrix(matrix, target));
    }
}